cmd/test.go

```diff
@@ -0,0 +1,290 @@
+// Copyright © 2022-2024 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/obolnetwork/charon/app/errors"
+)
+
+type testConfig struct {
+	OutputFormat string
+	OutputFile   string
+	Quiet        bool
+	TestCases    []string
+	Timeout      time.Duration
+}
+
+func newTestCmd(cmds ...*cobra.Command) *cobra.Command {
+	root := &cobra.Command{
+		Use:   "test",
+		Short: "Test subcommands provide test suite to evaluate current cluster setup",
+		Long:  `Test subcommands provide test suite to evaluate current cluster setup. Currently there is support for peer connection tests, beacon node and validator API.`,
+	}
+
+	root.AddCommand(cmds...)
+
+	return root
+}
+
+func bindTestFlags(cmd *cobra.Command, config *testConfig) {
+	cmd.Flags().StringVar(&config.OutputFile, "output-file", "", "File path to which output can be written.")
+	cmd.Flags().StringVar(&config.OutputFormat, "output-format", "json", "File format to which output is written. Flag --output-file is required.")
```

Comment:
What other formats you expect we may want to add?
I would go for the single format - TOML, which is both human and machine readable.

Reply:
Yes, TOML is one other on my list. Do you suggest we have single one and get rid of this env var?

Reply:
This will be much more user-friendly. IMO.
Operators (or DevOps) are very familiar with TOML/YAML via docker-compose experience.

Reply:
Changed to hardcoded TOML and renamed the flag as `--output-toml` to be a bit more clear when invoked.